docs/source/add_new_model.rst

```diff
@@ -0,0 +1,811 @@
+   `predict`, `evaluate`, `forward` or `__call__`. You don't want to debug a function that calls `forward` multiple
+   times, *e.g.* to generate text, like `autoregressive_sample`, `generate`.
+-  Try to separate the tokenization from the model's `forward` pass. If the original repository shows examples where
+   you have to input a string, then try to find out where in the forward call the string input is changed to input ids
+   and start from this point. This might mean that you have to possibly write a small script yourself or change the
+   original code so that you can directly input the ids instead of an input string.
+-  Make sure that the model in your debugging setup is **not** in training mode, which often causes the model to yield
+   random outputs due to multiple dropout layers in the model. Make sure that the forward pass in your debugging
+   environment is **deterministic** so that the dropout layers are not used. Or use `transformers.file_utils.set_seed`
+   if the old and new implementations are in the same framework.
+
+The following section gives you more specific details/tips on how you can do this for *brand_new_bert*.
+
+Implement BrandNewBert into 🤗 Transformers
+-----------------------------------------------------------------------------------------------------------------------
+
+Next, you can finally add code to 🤗 Transformers. Go into the clone of your 🤗 Transformers’ fork:
+
+::
+
+   cd transformers
+
+In the special case that you are adding a model whose architecture exactly matches the model architecture of an
+existing model you only have to add a conversion script as described in `this section <#write-a-conversion-script>`__.
+In this case, you can just re-use the whole model architecture of the already existing model.
+
+Otherwise, let’s start generating a new model with the amazing Cookiecutter!
+
+**Use the Cookiecutter to automatically generate the model’s code**
+
+To begin with head over to the `🤗 Transformers templates
+<https://github.com/huggingface/transformers/tree/master/templates/adding_a_new_model>`__ to make use of our
+``cookiecutter`` implementation to automatically generate all the relevant files for your model. Again, we recommend
+only adding the PyTorch version of the model at first. Make sure you follow the instructions of the ``README.md`` on
+the `🤗 Transformers templates <https://github.com/huggingface/transformers/tree/master/templates/adding_a_new_model>`__
+carefully.
+
+**Open a Pull Request on the main ``huggingface/transformers`` repo**
+
+Before starting to adapt the automatically generated code, now is the time to open a “Work in progress (WIP)” pull
+request, *e.g.* “[WIP] Add *brand_new_bert*”, in 🤗 Transformers so that you and the Hugging Face team can work
+side-by-side on integrating the model into 🤗 Transformers.
+
+You should do the following:
+
+1. Commit the automatically generated code:
```

Comment:
Unless I missed something you're recommending they work on their `master` branch which is a bad idea. 

should we have recommended to immediately create a branch once they cloned their repo? or here is good too.